packages/expo-in-app-purchases/src/InAppPurchases.types.ts

```diff
@@ -0,0 +1,63 @@
+export type ValidItemType = 'inapp' | 'subs';
```

Comment:
Use an enum so that (a) we don't directly expose the native values and (b) use more descriptive names:

```suggestion
export enum IAPItemType = {
  PURCHASE = 0,
  SUBSCRIPTION = 1,
};
```

Reply:
In this case, the reason I use string literals is because that's what the native code returns. See this TS definition:
```typescript
export interface ItemDetails {
  ...
  type: ValidItemType,
  ...
}
```
I don't actually need to export it though since it's not used outside of this file. Maybe I can just remove this type and use string literals for the `ItemDetails.type` attribute? E.g. something like this:
```typescript
export interface ItemDetails {
  ...
  type: 'inapp' | 'subs',
  ...
}
```

Reply:
The point of (a) above is intentionally not to expose the native values and to use Expo-defined strings instead so that the cross-platform API is more independent from any one platform.

Reply:
Ok that makes sense. So do you want me to change the values natively then too? Because the string literals are currently being returned. Or is just removing the type ok too?

Reply:
The goal is to define one type that is common to all platforms. The native values don't actually matter -- they can be numbers, strings, whatever -- as long as they map to the values defined by the single cross-platform interface.

Reply:
Oh ok and is it bad practice to model an aspect of the cross-platform interface in TS after one platform? In this case I'm borrowing the terminology from Android and having iOS match it. Occasionally, I do the inverse. Do we have a convention for this?